### ISSUE TITLE
Consulta TP

### ISSUE THREAD

**Issue by ssommerr** [PI-ITBA/2024_02#203]
Hola, como andan? Queria consultar si se podria suponer que la mayoria de las agencias van a tener alguna multa todos los años a partir del  1900. Y aparte consultar si es preferible que las funciones sean eficientes a rasgos de memoria o usar menos memoria pero tener menos eficiencia?

**Comment by marcelogarberoglio** [comment]
Buenas. Tengan en cuenta que las estadísticas deben funcionar con archivos vacíos (sólo los headers pero sin datos), archivos muy grandes y también archivos pequeños de prueba. Si tu pregunta de "suponer que la mayoría de las agencias..." apunta a eficiencia te diría que sí, que es esperable, pero debe funcionar también para otros casos.
En cuanto a qué privilegiar, que sean eficientes en cuanto al tiempo de ejecución. Eso no significa que puede desperdiciar memoria, repasen el ejercicio 10 del TP 11

**Comment by lcrosta** [comment]
hola marcelo, por ejemplo si no hay multas, no hay agencias, las estadisticas no mostrarian nada, pero el codigo deberia devolver el archivo csv con solo los headers de las querys?

**Comment by ssommerr** [comment]
Una consulta mas 30 caracteres es considerado corto, como un nombre?

**Comment by lcrosta** [comment]
otra consulta, investigamos sobre la lectura de archivos y encontramos que si eran muchos datos era mejor una lectura de a bloques, es necesario esto? o si lo hacemos linea por linea estaria bien?

**Comment by marcelogarberoglio** [comment]
> hola marcelo, por ejemplo si no hay multas, no hay agencias, las estadisticas no mostrarian nada, pero el codigo deberia devolver el archivo csv con solo los headers de las querys?

Si no hay multas se deben crear todos los archivos sólo con el encabezado

**Comment by marcelogarberoglio** [comment]
> otra consulta, investigamos sobre la lectura de archivos y encontramos que si eran muchos datos era mejor una lectura de a bloques, es necesario esto? o si lo hacemos linea por linea estaria bien?

No hace falta que hagan nada especial con la lectura de los archivos. Si usan las funciones de la biblioteca estándar no es que por cada lectura accede al disco sino que hace una lectura "bufferizada". Por ejemplo si uds usan fgetc para leer el primer carácter se lee un bloque del disco, lo almacena y retorna el primer byte. Si luego hacen otro fgetc para leer el siguiente char no accede al disco sino que ya lo tiene en la memoria, recién cuando leyeron todo el bloque en memoria y quieren leer uno más vuelve a acceder al disco. Así que da lo mismo si leen de a byte, línea o bloque, siempre y cuando lo hagan en forma secuencial.

**Comment by ssommerr** [comment]
> > Una consulta mas 30 caracteres es considerado corto, como un nombre?
> 
> En el enunciado dice cuáles son las longitudes máximas de los nombres para cada ciudad

Claro, pero estaria mal usar un strlen en ese caso? O nos guiamos de la otra forma?

**Comment by marcelogarberoglio** [comment]
Salvo por el tema de validar malloc, calloc, realloc, para todo lo demás deben guiarse por los criterios vistos durante el cuatrimestre.
